core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1223,6 +1232,9 @@ private[spark] class BlockManager(
   }
 
   def stop(): Unit = {
+    if ((blockManagerId ne null) && blockManagerId.isDriver) {
```

Comment:
elsewhere in Spark we just do `if (blockManagerId != null && blockManagerId.isDriver)`